[Messenger] Add a Doctrine transport (#29007)
symfony/symfony: src/Symfony/Component/Messenger/CHANGELOG.md

CHANGELOG
=========

4.3.0
-----

>  * Add a `Doctrine` transport

Reviewer: Could we hint the DNS too `doctrine://`?

 * [BC BREAK] The Amqp Transport now automatically sets up the exchanges
   and queues by default. Previously, this was done when in "debug" mode
   only. Pass the `auto-setup` connection option to control this.

>  * Add a `Doctrine` transport using `doctrine://default` DSN (this use the `default` Doctrine entity manager configured)

Reviewer 2: typo: `this uses`

Reviewer 2: I would remove "configured"

   and queues by default. Previously, this was done when in "debug" mode
   only. Pass the `auto-setup` connection option to control this.

>  * Add a `Doctrine` transport using `doctrine://<connection_name>` DSN

Reviewer 3: Can you add a concrete example?

 * `ReceiverInterface` and its implementations have been moved to the `Transport\Receiver` sub-namespace
 * `ActivationMiddlewareDecorator` has been renamed `ActivationMiddleware`
 * `AllowNoHandlerMiddleware` has been removed in favor of a new constructor argument on `HandleMessageMiddleware`

>  * Add a `Doctrine` transport

Reviewer 4: should be on 4.3.0

 * [BC BREAK] The Amqp Transport now automatically sets up the exchanges
   and queues by default. Previously, this was done when in "debug" mode
   only. Pass the `auto_setup` connection option to control this.

>  * Add a `Doctrine` transport using `doctrine://default` DSN (this uses the `default` Doctrine entity manager)

Reviewer 3: ```suggestion
 * Added a Doctrine transport. For example, the `doctrine://default` DSN uses
   the `default` Doctrine entity manager.
```
